### PR TITLE
Improve error message for wrong struct initialization

### PR DESCRIPTION
**Pull request by federicobond**

### CODE REVIEW

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -1551,8 +1551,14 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 
 	if (!functionType->takesArbitraryParameters() && parameterTypes.size() != arguments.size())
 	{
+		string kindStr;
```
Comment (axic) [inline libsolidity/analysis/TypeChecker.cpp]: Why not call it `kind`?

Alternatively have a `bool isStructConstructorCall` and have a ternary operator in `msg`.

Comment (federicobond) [reply libsolidity/analysis/TypeChecker.cpp]: Yes, the boolean makes more sense, I hadn't given it much thought really.

**Comment by axic** [comment]
Also please amend the changelog.

**Comment by federicobond** [comment]
Updated.

File: Changelog.md [hunk 2]
```diff
@@ -9,6 +9,7 @@ Bugfixes:
  * Standard JSON: Populate the ``sourceLocation`` field in the error list.
  * Type Checker: Suggest the experimental ABI encoder if using ``struct``s as function parameters
    (instead of an internal compiler error).
+ * Type Checker: Improve error message for wrong struct initialization.
```
Comment (axic) [inline Changelog.md:12]: Nitpicking: this would be a feature? :)

Comment (federicobond) [reply Changelog.md:12]: It’s under the Bugfixes title.

Comment (axic) [reply Changelog.md:12]: I meant its more like a feature, but it doesn't really makes much of a difference.

**Review by axic — APPROVED** [review]

File: Changelog.md (hunk 2 above)
Comment (federicobond) [reply Changelog.md:12]: Yeah, not worth it.

**Pull request merged**